zend_abstract_interface/hook/hook.c

```diff
@@ -58,6 +59,19 @@ __thread HashTable zai_hook_request_classes;
 // if indirect, then it's pointing to some hashtable in zai_hook_request_functions/classes
 __thread HashTable zai_hook_resolved;
 
+// zai_hook_inheritors is a map of persistent class entries (interfaces and abstract classes) to a list of persistent class entries
+static HashTable zai_hook_static_inheritors;
```

Comment:
question: "persistent classes" as in preloaded? Like: which classes survive a request shutdown? Oh, maybe builtin classes?

Reply:
Yes, persistent classes include preloaded (there's also a comment in line 928) - builtin classes are "internal".